### ISSUE TITLE
ERROR Error: Uncaught (in promise): [object Event] 

### ISSUE THREAD

**Issue by vinayistar** [niklasvh/html2canvas#1593]
Im using this plugin angular 5 and I encounter this error every time when I take screenshot. Please suggest how to handle in Typescript not in Jquery all.
Thanks

**Comment by Warriormuffin** [comment]
I run into this same issue. I have never been able to fix it.  

<img width="543" alt="screen shot 2018-07-12 at 12 38 45 pm" src="https://user-images.githubusercontent.com/19913775/42652746-94ded2c4-85d0-11e8-9e0f-40d14b9ea07d.png">

**Comment by rscherer** [comment]
I've implemented html2canvas on a Angular 6 application and it's working as expected, with the following code:

`import * as html2canvas from 'html2canvas';

const element = document.getElementById('filters'); 
html2canvas(element).then((canvas) => { 
	const imgData = canvas.toDataURL('image/png'); 
	const link = document.createElement('a'); 
	link.download = 'my-image-name.jpeg'; 
	link.href = imgData; link.click(); 
});`

Since the element 'filters' exists, there's no problem. Could u try it or provide some example of your code?

**Comment by vinayistar** [comment]
```
let element = document.querySelector(".some-area");
        html2canvas(element, { logging: false, useCORS: true }).then(canvas => {
          var imgData = canvas.toDataURL("image/png");
          //Error comes after above statement I think so.
        });
```
But I have tried with getElementByClassName(), Its throws error. How its working for you?

**Comment by Warriormuffin** [comment]
html2canvas(document.getElementById('image-area'), {
             useCORS: true,
             logging: false,
             scale: 2.5
            }).then(canvas => {

**Comment by vinayistar** [comment]
Is it working with document.getElementById ??

**Comment by Warriormuffin** [comment]
Getting by class name doesn't work at all. Getting by Id works but throws that error and can stall anywhere between 1 and 10 seconds.

**Comment by vinayistar** [comment]
Im using `querySelector` as per documentation. But showing error.

**Comment by Warriormuffin** [comment]
`querySelector ` still throws the same error for me. What version of angular are you running. My guess it is something to do with our version.

**Comment by vinayistar** [comment]
Im using AngularCLI 1.6.6

**Comment by Warriormuffin** [comment]
You find any way to fix this?s

**Comment by LanderBeeuwsaert** [comment]
I have this also sometimes, not always. I'm not sure but I think this is linked to the fact that it happens when for that session, a browser pop-up is used to (in my application) login to firebase (using firebase authentication)

**Comment by adapptors** [comment]
I also have the same issue. I am using HTML2CANVAS and I see Error: Uncaught (in promise): [object Event]. I am using Ionic 3 and tested on Chrome, Mozilla Firefox and Internet Explorer.
Could you please help.

**Comment by marcodelphi** [comment]
Try to run outside angular zone:

this.zone.runOutsideAngular(() => {
//your code
})

**Comment by LarsBuur** [comment]
In our situation, the plugin works fine until we add an external script (google tag manager) which is async. Using this.ngZone.runOutsideAngular does not help.

**Comment by apitts** [comment]
Confirming this issue. For me, it was a Profitwell script that interfered. The Zendesk widget also interferes in the sense that it leads to multiple renderings but the end result is still OK. The solution for me was to add some logic to the Profitwell script tag that caused it not to be run on the route where html2canvas is being used. I wonder though whether a configurable timeout that gave these async scripts time to load might be a solution?

**Comment by adamszeibert** [comment]
I've found that in my case the error was coming from `iframeLoader` defined in https://github.com/niklasvh/html2canvas/blob/master/src/Clone.js at around line 572.

I could solve the issue by changing the code to `return Promise.resolve(cloneIframeContainer);`
I'm pretty sure this fix breaks a lots of other uses cases, but for what I needed it solved the issue.

**Comment by hubaozhang** [comment]
@adamszeibert Can you push your solution?

**Comment by adamszeibert** [comment]
@hubaozhang I've found a real solution. Here's the PR: https://github.com/niklasvh/html2canvas/pull/1681

**Comment by uditdubey** [comment]
Hello, i am using html2canvcas in my react v16 project. but throws error
below is my code,

componentWillReceiveProps(nextProps,nextState){
    if (this.props.imgFlag !== nextProps.imgFlag) {
      if (nextProps.imgFlag === true) {
        let myelement = $("#responsiveReactGridLayout");
        html2canvas(myelement, {
          background: '#ffffff',
          onrendered: function(canvas){
           
            var imgData = canvas.toDataURL('image/jpeg');
           
          }
        })
      }
    }
}

can anyone suggest what could be done?

thanks  !

**Comment by paranram** [comment]
html2canvas(document.getElementById('print-section')
        , { logging: false, useCORS: true, scale: 2.5 }).then(canvas => 
    {  
      var imgWidth = 208;   
      var imgHeight = canvas.height * imgWidth / canvas.width;  
  
      const contentDataURL = canvas.toDataURL('image/png');
      let pdf = new jspdf('p', 'mm', 'a4'); // A4 size page of PDF  
      var position = 0;  
      pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight);
      pdf.save('receipt.pdf');  
    });

**Comment by mumpo** [comment]
In my case, the error was being thrown because they were foreign iframes in my website (mainly for tracking purposes). html2canvas will try to clone these iframes and fail due to CORS restrictions.

This quick fix worked for me:

```javascript
html2canvas(element, {
    ignoreElements: function (element) {
        return element.nodeName === 'IFRAME';
    }
}).then(function (canvas) {
    ...
});
```

**Comment by LanderBeeuwsaert** [comment]
@mumpo 
yes, thank you very much! This solved it for me as well.

**Comment by creativeoneteam** [comment]
@mumpo That worked for me as well - THANK YOU!

**Comment by chenxueliang** [comment]
> Im using this plugin angular 5 and I encounter this error every time when I take screenshot. Please suggest how to handle in Typescript not in Jquery all.
> Thanks

I have a same problem. Have you solved it？

**Comment by chenxueliang** [comment]
I have a same problem. Who has a solution?

**Comment by chenxueliang** [comment]
![Uploading 企业微信截图_16184763417394.png…]()

**Comment by chenxueliang** [comment]
![企业微信截图_16184763417394](https://user-images.githubusercontent.com/20763655/114842053-0e7f8880-9e0b-11eb-83b8-869d6e8f78ee.png)

**Comment by niklasvh** [comment]
This should be fixed in v1.1.1. If you are still experiencing this, could you please make a repro of it
